src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -6566,7 +6578,8 @@ ProjectionNames QueryAnalyzer::resolveExpressionNode(QueryTreeNodePtr & node, Id
                     scope.scope_node->formatASTForErrorMessage());
 
             auto & table_node = node->as<TableNode &>();
-            result_projection_names.push_back(table_node.getStorageID().getFullNameNotQuoted());
+            if (result_projection_names.empty())
```

Comment:
Just curious: when can it be non-empty?

Reply:
In the scenario I saw, we can resolve identifier `x alias y` as TableNode (by alias), and in this case, at first, alias would be added to projection names, and then the storage name.